test/test_linalg.py

```diff
@@ -3632,18 +3632,31 @@ def invert_perm(p):
             return (d[0], d[1], d[2])
 
         def generate_inputs():
+            # transposed tensors
             for perm1, perm2 in itertools.product(itertools.permutations((0, 1, 2)), repeat=2):
                 b1 = make_tensor((num_batches, M, N), device, dtype, low=-1, high=1)
                 b2 = make_tensor((num_batches, N, O), device, dtype, low=-1, high=1)
                 b1 = b1.permute(perm1).contiguous().permute(invert_perm(perm1))
                 b2 = b2.permute(perm2).contiguous().permute(invert_perm(perm2))
                 yield b1, b2
+            # broadcasting tensors
             for b1, b2, b3, b4, b5, b6 in itertools.product((True, False), repeat=6):
                 shape1 = (num_batches if b1 else 1, M if b2 else 1, N if b3 else 1)
                 shape2 = (num_batches if b4 else 1, N if b5 else 1, O if b6 else 1)
                 b1 = make_tensor(shape1, device, dtype, low=-1, high=1).expand(num_batches, M, N)
                 b2 = make_tensor(shape2, device, dtype, low=-1, high=1).expand(num_batches, N, O)
                 yield b1, b2
+            # zero-sized tensors
+            bug = (self.device_type == 'cuda' and dtype == torch.half and torch.version.cuda is not None and
```

Comment:
wait, does cublas < 11 not handle empty tensors correctly? That's interesting. DO we have an issue for that to cite here?

Reply:
It was, but I don't think it is anymore after https://github.com/pytorch/pytorch/commit/9ae70284c75e82bda368352a148a77091a6742e3, I will update it.

Reply:
tests are passing, seem to be OK